packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -5,104 +5,184 @@
+}) as PolymorphicForwardRefComponent<'div', AlertOwnProps>;
+AlertComponent.displayName = 'Alert';
+
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+type AlertIconOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+
+  const { type } = useSafeContext(AlertContext);
+
+  if (!children) {
+    const StatusIcon = StatusIconMap[type];
+    return (
+      <Icon fill={type}>
+        <StatusIcon ref={ref} {...rest} />
+      </Icon>
+    );
+  }
+
+  return (
+    <Icon ref={ref} {...rest}>
+      {children}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<'span', AlertIconOwnProps>;
+AlertIcon.displayName = 'Alert.Icon';
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+const AlertMessage = polymorphic.span('iui-alert-message');
+AlertMessage.displayName = 'Alert.Message';
+
+// ----------------------------------------------------------------------------
+// Alert.Action component
+
+type AlertActionOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertAction = React.forwardRef((props, ref) => {
+  const { children, className, ...rest } = props;
+
+  return (
+    <Box
+      as={(!!props.href ? 'a' : 'button') as 'a'}
+      className={cx('iui-alert-link', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'a', AlertActionOwnProps>;
+AlertAction.displayName = 'Alert.Action';
+
+// ----------------------------------------------------------------------------
+// Alert.CloseButton component
+
+type AlertCloseButtonOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertCloseButton = React.forwardRef((props, ref) => {
+  const { children, ...rest } = props;
+
+  return (
+    <IconButton
+      styleType='borderless'
+      size='small'
+      aria-label='Close'
+      ref={ref}
+      {...rest}
+    >
+      {children ?? <SvgCloseSmall />}
+    </IconButton>
+  );
+}) as PolymorphicForwardRefComponent<'button', AlertCloseButtonOwnProps>;
+AlertCloseButton.displayName = 'Alert.CloseButton';
+
+/**
+ * A small box to quickly grab user attention and communicate a brief message
+ * @example
+ * <Alert>
+ *  <Alert.Message>This is an alert.</Alert.Message>
+ * </Alert>
+ * <Alert type='informational'>
+ *  <Alert.Icon />
+ *  <Alert.Message>This is an informational alert.</Alert.Message>
+ * </Alert>
+ * <Alert type='positive'>
+ *  <Alert.Icon>
+ *    <SvgSmileyHappy />
+ *  </Alert.Icon>
+ *  <Alert.Message>
+ *    This is an alert.
+ *    <Alert.Action>This is clickable text.</Alert.Action>
+ *  </Alert.Message>
+ *  <Alert.CloseButton onClick={action('Close!')}>
+ *    <Icon>
+ *      <SvgCollapse />
+ *    </Icon>
+ *  </Alert.CloseButton>
```

Comment:
Each example should get a separate `@example` tag.
```suggestion
 * @example
 * <Alert>
 *  <Alert.Message>This is an alert.</Alert.Message>
 * </Alert>
 *
 * @example
 * <Alert type='informational'>
 *  <Alert.Icon />
 *  <Alert.Message>This is an informational alert.</Alert.Message>
 *  <Alert.CloseButton onClick={() => {}} />
 * </Alert>
 *
 * @example
 * <Alert type='positive'>
 *  <Alert.Icon>
 *    <SvgSmileyHappy />
 *  </Alert.Icon>
 *  <Alert.Message>
 *    This is an alert.
 *    <Alert.Action>This is clickable text.</Alert.Action>
 *  </Alert.Message>
 *  <Alert.CloseButton onClick={() => {}}>
 *    <SvgCollapse />
 *  </Alert.CloseButton>
 * </Alert>
```

Reply:
This has been fixed

Reply:
take a closer look at my suggestion, there are some more suggestions there - e.g. the addition of `Alert.CloseButton` in second example, the removal of `Icon` and the removal of `action`

Reply:
Good catch. `Alert.CloseButton` has been added to the second example and `Icon` and `action` have been removed